src/states/node.rs

```diff
@@ -902,11 +921,12 @@ impl Node {
                 } = hop_msg;
                 self.handle_signed_message(content, route, hop_name, &sent_to)
             }
-            Err(RoutingError::ExceedsRateLimit) => {
+            Err(RoutingError::ExceedsRateLimit(hash)) => {
```

Comment:
just wondering : `hash` is in content already, so why pass it down to rate_limiter then get it back (and rate_limiter is not using hash)
can we do : 
```rust
Err(RoutingError::ExceedsRateLimit) => {
  if let MessageContent::UserMessagePart { hash, .. } = hop_msg.content.routing_message().content {
    ..
  } else {
    ..
  }
}
```
so we don't need to modify RoutingError and pass hash around ?